tests/sphinx/test_github_links.py

```diff
@@ -0,0 +1,188 @@
+import pathlib
+import sys
+
+from django.test import SimpleTestCase
+
+
+class GitHubLinkTests(SimpleTestCase):
+    @classmethod
+    def setUpClass(cls):
+        # The file implementing the code under test cannot be imported through
+        # standard means, so include its parent in the pythonpath for the
+        # duration of the tests.
+        cls.ext_path = str((pathlib.Path(__file__).parents[2] / "docs/_ext").resolve())
+        sys.path.insert(0, cls.ext_path)
+        cls.addClassCleanup(sys.path.remove, cls.ext_path)
+        cls.addClassCleanup(sys.modules.pop, "github_links", None)
+
+    @property
+    def github_links(self):
+        # The import must happen after setUpClass, so it can't be imported at
+        # the top of the file. A property is used to avoid the import in every test.
+        # Linters/IDEs may not be able to detect this as a valid import.
+        import github_links
+
+        return github_links
+
+    def test_code_locator(self):
+        locator = self.github_links.CodeLocator.from_code(
+            """
+from a import b, c
+from .d import e, f as g
+
+def h():
+    pass
+
+class I:
+    def j(self):
+        pass"""
+        )
+
+        self.assertEqual(locator.node_line_numbers, {"h": 5, "I": 8, "I.j": 9})
+        self.assertEqual(locator.import_locations, {"b": "a", "c": "a", "e": ".d"})
+
+    def test_module_name_to_file_path_package(self):
+        path = self.github_links.module_name_to_file_path("django")
+
+        self.assertEqual("/".join(path.parts[-2:]), "django/__init__.py")
```

Comment:
Since this pattern is heavily used across many tests, what do you think if we add a helper `last_n_parts` or similar so:
```python
def last_n_parts(path, n):
    return "/".join(path.parts[-n:])
```

And then:
```python
        self.assertEqual(last_n_parts(path, 2), "django/__init__.py")
```